### PR TITLE
feat: custom json request detect

### PR DESCRIPTION
**Pull request by fengmk2**
In special case, we let `/*.json` request as json request,
and need to parse request body with json format.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ var copy = require('copy-to');
 
 module.exports = function (opts) {
   opts = opts || {};
+  var detectJSON = opts.detectJSON;
+  opts.detectJSON = undefined;
```
Comment (dead-horse) [inline index.js]: why set `opts.detectJSON` to undefined?

Comment (fengmk2) [inline index.js]: stop pass it to `parse.json(opts)` and `parse.form(opts)`

**Comment by fengmk2** [comment]
publish minor version

**Comment by dead-horse** [comment]
1.4.0

File: index.js (hunk 1 above)
Comment (popomore) [inline index.js]: why not `delete` it?

Comment (fengmk2) [inline index.js]: https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#521-objects-that-are-in-hash-table-mode-aka-normalized-objects-dictionary-mode---objects-who-have-a-hash-table-as-a-backing-data-structure-are-not-simple-enumerables

Comment (luckydrq) [inline index.js]: :+1:

**Pull request merged**